crates/floresta-wire/src/p2p_wire/sync_node.rs

```diff
@@ -143,7 +162,11 @@ where
             try_and_log!(self.request_blocks(vec![block]).await);
         }
     }
-
+    /// Process a block received from a peer.
+    /// This function removes the received block from the inflight requests and inserts in its own blocks map.
+    /// It then processes the block and its proof, and connects it to the chain.
+    ///
+    /// This function bans the peer if the block is invalid, and requests a new block from another peer.
```

Comment:
If the block is invalid, we don't request from anyone else. The chain as a whole is invalid.